### ISSUE TITLE
Bug starting with 11.0.7

### ISSUE THREAD

**Issue by zaphon** [Dathwada/handynotes-travelguide#56]
Since the launch of 11.0.7 I get 2 errors on login. They are triggered by a change in the VignetteDataProvider (it takes a vignetteInfo parameter to GetPinTemplate that the other DataProviders do not).

```
5x ...zard_SharedMapDataProviders/VignetteDataProvider.lua:14: attempt to index local 'vignetteInfo' (a nil value)
[string "@Blizzard_SharedMapDataProviders/VignetteDataProvider.lua"]:14: in function `GetPinTemplate'
[string "@HandyNotes_TravelGuide/core/handler.lua"]:116: in main chunk

Locals:
self = <table> {
 uniqueVignettesGUIDs = <table> {
 }
 owningMap = WorldMapFrame {
 }
 uniqueVignettesPins = <table> {
 }
 vignetteGuidsToPins = <table> {
 }
}
vignetteInfo = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index local 'vignetteInfo' (a nil value)"
```

and

```
9x ...rfaceHandyNotes_TravelGuide/core/handler.lua:101: attempt to index field 'db' (a nil value)
[string "@HandyNotes_TravelGuide/core/handler.lua"]:101: in function <...rfaceHandyNotes_TravelGuide/core/handler.lua:100>
[string "=[C]"]: in function `RefreshAllData'
[string "@Blizzard_MapCanvas/MapCanvas_DataProviderBase.lua"]:76: in function `OnMapChanged'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:662: in function <...aceBlizzard_MapCanvas/Blizzard_MapCanvas.lua:661>
[string "=[C]"]: in function `secureexecuterange'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:667: in function `OnMapChanged'
[string "@Blizzard_WorldMap/Blizzard_WorldMap.lua"]:328: in function <...rfaceBlizzard_WorldMap/Blizzard_WorldMap.lua:327>
[string "=[C]"]: in function `OnMapChanged'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:61: in function `SetMapID'
[string "@Blizzard_WorldMap/Blizzard_WorldMap.lua"]:344: in function <...rfaceBlizzard_WorldMap/Blizzard_WorldMap.lua:337>
[string "=[C]"]: in function `Show'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:493: in function `SetUIPanel'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:336: in function `ShowUIPanel'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:174: in function <...ParentPanelManager/Mainline/UIParentPanelManager.lua:169>
[string "=[C]"]: in function `SetAttribute'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:893: in function <...ParentPanelManager/Mainline/UIParentPanelManager.lua:871>
[string "=[C]"]: in function `ShowUIPanel'
[string "@Blizzard_WorldMap/QuestLogOwnerMixin.lua"]:115: in function `SetDisplayState'
[string "@Blizzard_WorldMap/Blizzard_WorldMap.lua"]:169: in function <...rfaceBlizzard_WorldMap/Blizzard_WorldMap.lua:160>

Locals:
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index field 'db' (a nil value)"
ns = <table> {
 config = <table> {
 }
 DB = <table> {
 }
 locale = <table> {
 }
 constants = <table> {
 }
}
areaPoisToRemove = <table> {
 1 = 5846
 2 = 5847
 3 = 5848
 4 = 5873
 5 = 5874
 6 = 5875
 7 = 5876
 8 = 5877
 9 = 5878
 10 = 5879
 11 = 5880
 12 = 5892
 13 = 6014
 14 = 7340
 15 = 7335
 16 = 5843
 17 = 5844
 18 = 5845
 19 = 5883
 20 = 5884
 21 = 5885
 22 = 5886
 23 = 5887
 24 = 5888
 25 = 5890
 26 = 6015
 27 = 6138
 28 = 7339
 29 = 7341
 30 = 5881
 31 = 5882
 32 = 7017
 33 = 7019
 34 = 7020
 35 = 7944
 36 = 7945
 37 = 7959
 38 = 7960
 39 = 7961
 40 = 8001
 41 = 8002
 42 = 8003
 43 = 8004
 44 = 8006
 45 = 8009
 46 = 8010
 47 = 8171
}
addon = <table> {
 baseName = "HandyNotes_TravelGuide"
 description = "Displays the portal, zeppelin and boat locations on the world map and minimap."
 pluginName = "TravelGuide"
 modules = <table> {
 }
 defaultModuleLibraries = <table> {
 }
 enabledState = true
 defaultModuleState = true
 name = "HandyNotes_TravelGuide"
 orderedModules = <table> {
 }
}

```

I tried to find a quick fix for this, and I don't see anything obvious.

**Comment by Dathwada** [comment]
Hi, I will take a look at this later today (german time) when I get back from work.

**Comment by Dathwada** [comment]
I just released v2.21.0, it is available here on [GitHub](https://github.com/Dathwada/handynotes-travelguide/releases/tag/Retail-v2.21.0) and should be available on [CurseForge](https://legacy.curseforge.com/wow/addons/handynotes-travelguide/files/6002677) or [Wago](https://addons.wago.io/addons/handynotes-travelguide/versions?stability=stable) in a few minutes.
